src/config/parser.rs

```diff
@@ -6,6 +6,7 @@ use crate::handler::{file_path, FileType};
 use crate::models::parser_models::{AggregatorConfig, RateLimiter, Style};
 use log::LevelFilter;
 use mlua::Lua;
+use reqwest::Proxy;
```

Comment:
```suggestion
#[cfg(feature = "socks")]
use reqwest::Proxy;
```

I would suggest having this behind a `cfg` feature macro as this will only add the following code to the binary if the `socks` feature has been enabled during build otherwise it will not include it thus reducing binary size and removing unneeded code. 🙂